### PR TITLE
add more tests for the GQ private transaction bloom filters

### PR DESCRIPTION
**Pull request by pinges**
Signed-off-by: Stefan Pingel <stefan.pingel@consensys.net>

### CODE REVIEW

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/query/GQPrivateTxBloomBlockchainQueriesTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  }
+
+  /**
+   * Tests whether block headers containing private blooms match. The resulting list of
+   * LogWithMetadata would be empty, because the mocked blockchain does not return any receipts, but
+   * we do check that the methods on the blockchain are actually called that would be called if the
+   * block header matches the bloom filter.
+   */
+  @Test
+  public void uncachedUncachedSeamTest() {
```
Comment (macfarla) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/query/GQPrivateTxBloomBlockchainQueriesTest.java:118]: is there logic behind this test name?

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/query/GQPrivateTxBloomBlockchainQueriesTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GQPrivateTxBloomBlockchainQueriesTest {
```
Comment (macfarla) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/query/GQPrivateTxBloomBlockchainQueriesTest.java:53]: I know it's long but prob should expand the shortcuts
GoQuorumPrivateTransactionBloomBlockchainQueriesTest

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/query/cache/GQPrivateTransactionLogBloomCacherTest.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@SuppressWarnings("unused")
+@RunWith(MockitoJUnitRunner.class)
+public class GQPrivateTransactionLogBloomCacherTest {
```
Comment (macfarla) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/query/cache/GQPrivateTransactionLogBloomCacherTest.java:61]: GoQuorumPrivateTransactionLogBloomCacherTest

**Review by macfarla — APPROVED** [review]
LGTM. some minor comments

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/query/cache/GQPrivateTransactionLogBloomCacherTest.java (hunk 3 above)
Comment (pinges) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/query/cache/GQPrivateTransactionLogBloomCacherTest.java]: Done

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/query/GQPrivateTxBloomBlockchainQueriesTest.java (hunk 2 above)
Comment (pinges) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/query/GQPrivateTxBloomBlockchainQueriesTest.java]: Done

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/query/GQPrivateTxBloomBlockchainQueriesTest.java (hunk 1 above)
Comment (pinges) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/query/GQPrivateTxBloomBlockchainQueriesTest.java]: fixed

**Pull request merged**